lib/manageiq/messaging/queue.rb

```diff
@@ -3,15 +3,20 @@ module Messaging
     class Queue
       include Common
 
-      def self.publish(client, options)
+      def self.publish(client, options, &block)
         assert_options(options, [:message, :service])
 
         address, headers = queue_for_publish(options)
         headers[:sender] = options[:sender] if options[:sender]
         headers[:message_type] = options[:message] if options[:message]
         headers[:class_name] = options[:class_name] if options[:class_name]
+        headers[:correlation_id] = Time.now.to_i.to_s if block_given?
```

Comment:
Changed `correlation_id` to `correlation_ref` where possible. Here the keyword is reserved by ActiveMQ.